emerge/output/html/resources/js/emerge_graph.js

```diff
@@ -357,117 +357,111 @@ function stringIncludedInNodeContributors(string, node) {
 function drawNodeToolTip(text, xPos, yPos, nodeMetrics) {
     
     // $('#overallStatisticsModal').modal('show');
+
+    // Calculate scale factor to maintain fixed size when d3 canvas zoom changes.
+    const zoomTransform = d3.zoomTransform(context.canvas);
+    const scaleFactor = 1 / zoomTransform.k;
+
+    // Scale linewidth and font so they're unchanged when zooming:
+    context.lineWidth = scaleFactor;
+    const scaledFontSize = 14 * scaleFactor;
+    context.font = scaledFontSize + 'px Helvetica';
+
+    // Extract code metrics and semantic keywords from `nodeMetrics` object:
+    const codeMetricLabels = formatCodeMetricLabels(nodeMetrics);
+    const semanticKeywords = formatSemanticKeywords(nodeMetrics);
```

Comment:
The second commit in this PR refactors the metric label logic to pre-compute the formatted labels once and use those pre-computed arrays to calculate both the `textBoxWidth` and to render the text. I would imagine the additional memory use would be small compared to the rest of the data, but it's possible for large graphs, this might be undesirable. I'm happy to revert if it's an issue.